### PR TITLE
Fix connect upstream test in nested edge.

### PR DESCRIPTION
**Pull request by huguesBouvier**
Connect upstream test will now use parent hostname instead of iothubhostname when nested.

Decorate error message in nested to help user troubleshoot.

Test:
Tested happy path on lvl5(non nested)
Tested happy path on lvl4 and made sure it was connecting to parent.
Removed trust bundle from store and correctly got the following error message:
host can connect to and perform TLS handshake with iothub AMQP port - Error
    Make sure the parent device is reachable using 'curl https://parenthostname'. Make sure the the trust bundle has been added to the trusted store: sudo cp <path>/azure-iot-test-only.root.ca.cert.pem /usr/local/share/ca-certificates/azure-iot-test-only.root.ca.cert.pem.crt
                sudo update-ca-certificates

### CODE REVIEW

File: aziotctl/src/internal/check/checks/host_connect_iothub.rs [hunk 1]
```diff
@@ -68,17 +68,35 @@ impl HostConnectIotHub {
     ) -> Result<CheckResult> {
         use aziot_identityd_config::ProvisioningType;
 
-        let iothub_hostname = match &shared.cfg.iothub_hostname {
-            Some(s) => s,
+        let identityd_config = unwrap_or_skip!(&cache.cfg.identityd);
+
+        let (iothub_hostname, nested) = match &shared.cfg.iothub_hostname {
+            Some(s) => (s, false),
```
Comment (daprilik) [inline aziotctl/src/internal/check/checks/host_connect_iothub.rs:74]: this raises the question of whether or not we should add a `--gateway-hostname` flag to compliment the existing `--iothub-hostname` flag that already exists.

Obviously not part of this PR, but this is nonetheless something we should think about.

File: aziotctl/src/internal/check/checks/host_connect_iothub.rs [hunk 2]
```diff
@@ -68,17 +68,35 @@ impl HostConnectIotHub {
...
-                    } => iothub_hostname,
+                    } => {
+                        // use `local_gateway_hostname` if available
+                        (
+                            identityd_config
+                                .provisioning
+                                .local_gateway_hostname
+                                .as_ref()
+                                .unwrap_or(iothub_hostname),
+                            true,
```
Comment (daprilik) [inline aziotctl/src/internal/check/checks/host_connect_iothub.rs:87]: shouldn't this be `identityd_config.provisioning.local_gateway_hostname.is_some()`?

Comment (huguesBouvier) [reply aziotctl/src/internal/check/checks/host_connect_iothub.rs]: Good catch it was a bug

**Review by daprilik — APPROVED** [review]
